Error boundaries docs page translation (#104)
reactjs/pl.reactjs.org: content/docs/error-boundaries.md

---
id: error-boundaries
title: Granice błędów
permalink: docs/error-boundaries.html
---

> W przeszłości, błędy JavaScriptowe w komponentach psuły wewnętrzny stan Reacta i [wywoływały](https://github.com/facebook/react/issues/4026) [enigmatyczne](https://github.com/facebook/react/issues/6895) [błędy](https://github.com/facebook/react/issues/8579) w kolejnych renderach. Były one następstwem wcześniejszego błędu w kodzie aplikacji, którego React nie był w stanie obsłużyć.

Reviewer: ```suggestion
W przeszłości, błędy JavaScriptowe wewnątrz komponentów uszkadzały wewnętrzny stan Reacta i [wywoływały](https://github.com/facebook/react/issues/4026) [tajemnicze](https://github.com/facebook/react/issues/6895) [błędy](https://github.com/facebook/react/issues/8579) w kolejnych renderach. Były one następstwem wcześniejszego błędu w kodzie aplikacji, jednakże React nie dostarczał żadnego rozwiązania, aby je właściwie obsłużyć wewnątrz komponentów oraz nie mógł powrócić do życia po ich wystąpieniu.
```

> W przeszłości, błędy JavaScriptowe w komponentach psuły wewnętrzny stan Reacta i [wywoływały](https://github.com/facebook/react/issues/4026) [enigmatyczne](https://github.com/facebook/react/issues/6895) [błędy](https://github.com/facebook/react/issues/8579) w kolejnych renderach. Były one następstwem wcześniejszego błędu w kodzie aplikacji, którego React nie był w stanie obsłużyć.

Reviewer 2: ```suggestion
W przeszłości, błędy javascriptowe wewnątrz komponentów uszkadzały wewnętrzny stan Reacta i [wywoływały](https://github.com/facebook/react/issues/4026) [tajemnicze](https://github.com/facebook/react/issues/6895) [błędy](https://github.com/facebook/react/issues/8579) w kolejnych renderowaniach. Były one następstwem wcześniejszego błędu w kodzie aplikacji, jednakże React nie dostarczał żadnego rozwiązania pozwalającego na właściwe ich obsłużenie wewnątrz komponentów oraz nie potrafił odtworzyć aplikacji po ich wystąpieniu.
```

> ## Przedstawiamy Granice Błędów {#introducing-error-boundaries}

Reviewer 2: ```suggestion
## Przedstawiamy granice błędów {#introducing-error-boundaries}
```
W języku polskim słowa w tytułach zapisujemy w inny sposób niż w angielskim.

> W przeszłości, błędy javascriptowe wewnątrz komponentów uszkadzały wewnętrzny stan Reacta i [wywoływały](https://github.com/facebook/react/issues/4026) [tajemnicze](https://github.com/facebook/react/issues/6895) [błędy](https://github.com/facebook/react/issues/8579) w kolejnych renderowaniach. Były one następstwem wcześniejszego błędu w kodzie aplikacji, jednakże React nie dostarczał żadnego rozwiązania pozwalającego na właściwe ich obsłużenie wewnątrz komponentów oraz nie potrafił odtworzyć aplikacji po ich wystąpieniu.

Reviewer 2: ```suggestion
W przeszłości błędy javascriptowe wewnątrz komponentów uszkadzały wewnętrzny stan Reacta i [wywoływały](https://github.com/facebook/react/issues/4026) [tajemnicze](https://github.com/facebook/react/issues/6895) [błędy](https://github.com/facebook/react/issues/8579) w kolejnych renderowaniach. Były one następstwem wcześniejszego błędu w kodzie aplikacji, jednakże React nie dostarczał żadnego rozwiązania pozwalającego na właściwe ich obsłużenie wewnątrz komponentów oraz nie potrafił odtworzyć aplikacji po ich wystąpieniu.
```

> Błąd JavaScriptu w jednej z części interfejsu użytkownika (UI) nie powinien psuć całej aplikacji. Aby rozwiązać ten problem, w React 16 wprowadziliśmy koncepcję granic błędów (ang. error boundary).

Reviewer: ```suggestion
Błąd w kodzie JavaScript, występujący w jednej z części interfejsu użytkownika (UI) nie powinien psuć całej aplikacji. Aby rozwiązać ten problem, w React 16 wprowadziliśmy koncepcję granic błędów (ang. *error boundary*).
```

> Granice błędów to komponenty Reactowe, które **wyłapują, logują i wyświetlają błędy JavaScriptu we wszystkich swoich podrzędnych komponentach, a także wyświetlają zastępcze fragmenty interfejsu** zamiast pokazywać te niepoprawnie działające. Granice błędów działają podczas renderowania, w metodach cyklu życia komponentów, a także w konstruktorach całego podrzędnego drzewa komponentów.

Reviewer: ```suggestion
Granice błędów to komponenty Reacta, które **przechwytują błędy JavaScript, występujące gdziekolwiek wewnątrz drzewa komponentów ich dzieci, następnie logują je i wyświetlają rezerwowe UI,** zamiast pokazywać te niepoprawnie działające. Granice błędów przechwytują błędy występujące podczas renderowania, w metodach cyklu życia komponentów, a także w konstruktorach całego podrzędnego im drzewa komponentów.
```

Reviewer 2: `komponenty reactowe`, a nie `Reacta`.

> Granice błędów to komponenty Reactowe, które **wyłapują, logują i wyświetlają błędy JavaScriptu we wszystkich swoich podrzędnych komponentach, a także wyświetlają zastępcze fragmenty interfejsu** zamiast pokazywać te niepoprawnie działające. Granice błędów działają podczas renderowania, w metodach cyklu życia komponentów, a także w konstruktorach całego podrzędnego drzewa komponentów.

Reviewer 2: ```suggestion
Granice błędów to komponenty reactowe, które **przechwytują błędy javascriptowe występujące gdziekolwiek wewnątrz drzewa komponentów ich potomków, a następnie logują je i wyświetlają awaryjny interfejs UI**, zamiast pokazywać ten niepoprawnie działający. Granice błędów przechwytują błędy występujące podczas renderowania, w metodach cyklu życia komponentów, a także w konstruktorach całego podrzędnego im drzewa komponentów.
```

> Uwaga
>
> Granice błędów **nie obsługują** błędów w:
>

> > * Uchwytach zdarzeń (ang. event handlers) ([więcej](#how-about-event-handlers))

Reviewer 2: ```suggestion
> * Procedurach obsługi zdarzeń (ang. event handlers) ([informacje](#how-about-event-handlers))
```

> > * Asynchronicznym kodzie (np. w metodach: `setTimeout` lub w callbackach `requestAnimationFrame`)

Reviewer 2: ```suggestion
> * Asynchronicznym kodzie (np. w metodach: `setTimeout` lub w funkcjach zwrotnych `requestAnimationFrame`)
```

> * Komponentach renderowanych po stronie serwera
> * Błędach rzuconych w samych granicach błędów (a nie w ich podrzędnych komponentach)

> Aby komponent klasowy był granicą błędu musi zdefiniować jedną lub obie metody cyklu życia: [`static getDerivedStateFromError()`](/docs/react-component.html#static-getderivedstatefromerror) lub [`componentDidCatch()`](/docs/react-component.html#componentdidcatch). Należy używać `static getDerivedStateFromError()` do wyrenderowania zastępczego UI po rzuceniu błędu, a `componentDidCatch()`, aby zalogować informacje o błędzie.

Reviewer: ```suggestion
Aby komponent klasowy był granicą błędu, musi zdefiniować jedną lub obie metody cyklu życia: [`static getDerivedStateFromError()`](/docs/react-component.html#static-getderivedstatefromerror) lub [`componentDidCatch()`](/docs/react-component.html#componentdidcatch). Należy używać `static getDerivedStateFromError()` do wyrenderowania zastępczego UI po rzuceniu błędu, a `componentDidCatch()`, aby zalogować informacje o błędzie.
```

Reviewer 2: ```suggestion
Aby komponent klasowy stał się granicą błędu, musi definiować jedną lub obie metody cyklu życia: [`static getDerivedStateFromError()`](/docs/react-component.html#static-getderivedstatefromerror) i/lub [`componentDidCatch()`](/docs/react-component.html#componentdidcatch). Należy używać `static getDerivedStateFromError()` do wyrenderowania zastępczego UI po rzuceniu błędu, a `componentDidCatch()`, aby zalogować informacje o błędzie.
```

  }

  static getDerivedStateFromError(error) {

>     // Aktualizacja stanu - w następnym renderze zostanie pokazane UI zastępcze

Reviewer: ```suggestion
    // Zaktualizuj stan aby następny render pokazał zastępcze UI.
```

    return { hasError: true };
  }

  componentDidCatch(error, info) {

>     // componentDidCatch pozwala zalogować błąd do zewnętrznego serwisu

Reviewer: ```suggestion
    // Możesz także zalogować błąd do zewnętrznego serwisu raportowania błędów
```

    logErrorToMyService(error, info);
  }

  render() {
    if (this.state.hasError) {

>       // Jeżeli wystąpił błąd wyświetlamy dowolne zastępcze komponenty

Reviewer: ```suggestion
      // Możesz wyrenderować dowolny interfejs zastępczy.
```

}
```

Po zdefiniowaniu, granicy błędu można używać jak normalnego komponentu:

```js
<ErrorBoundary>
  <MyWidget />
</ErrorBoundary>
```

> Granice błędów działają jak bloki `catch {}` tyle, że dla komponentów. Można je zdefiniować tylko w komponentach klasowych. W praktyce w większości przypadków wystarczające będzie zdefiniowanie pojedynczego komponentu granicy błędu i używanie go w całej aplikacji.

Reviewer: ```suggestion
Granice błędów działają jak bloki `catch {}` w JavaScript, tyle, że dla komponentów. Można je zdefiniować tylko w komponentach klasowych. W praktyce, w większości przypadków wystarczające będzie zdefiniowanie pojedynczego komponentu granicy błędu i używanie go w całej aplikacji.
```

> Należy pamiętać, że **granice błędów wyłapują błędy w komponentach-dzieciach**. Nie są one jednak w stanie obsłużyć własnych błędów. W takim przypadku, jeżeli granica błędu nie będzie w stanie wyświetlić komunikatu błędu, zostanie on przekazany do kolejnej najbliższej granicy błędu wyżej w strukturze komponentów. Jest to zachowanie podobne do tego znanego z JavaScriptowych bloków `catch {}`.

Reviewer: ```suggestion
Należy pamiętać, że **granice błędów wyłapują błędy tylko w komponentach znajdujących się w drzewie komponentów niżej niż one same**. Nie są one jednak w stanie przechwytywać błędów występujących wewnątrz ich samych. W takim przypadku, jeżeli granica błędu nie będzie w stanie wyświetlić komunikatu błędu, zostanie on przekazany do kolejnej najbliższej granicy błędu wyżej w strukturze komponentów. Jest to zachowanie podobne do tego znanego z JavaScriptowych bloków `catch {}`.
```

Reviewer 2: ```suggestion
Należy pamiętać, że **granice błędów wyłapują błędy w komponentach potomnych**. Nie są one jednak w stanie obsłużyć własnych błędów. W takim przypadku, jeżeli granica błędu nie będzie w stanie wyświetlić zastępczego UI, błąd zostanie przekazany do kolejnej najbliższej granicy błędu powyżej w strukturze komponentów. Jest to zachowanie podobne do tego znanego z javascriptowych bloków `catch {}`.
```

## Demo {#live-demo}

> [Przykład tworzenia i używania granicy błędów](https://codepen.io/gaearon/pen/wqvxGa?editors=0010) z wykorzystaniem [Reacta 16](/blog/2017/09/26/react-v16.0.html).

Reviewer: ```suggestion
[Przykład tworzenia i użycia granicy błędów](https://codepen.io/gaearon/pen/wqvxGa?editors=0010) z wykorzystaniem [Reacta 16](/blog/2017/09/26/react-v16.0.html).
```

## Gdzie umiejscowić granice błędów {#where-to-place-error-boundaries}

> To jak bardzo szczegółowo zostanie pokryty kod za pomocą granic błędów jest kwestią preferencji. Możliwe jest na przykład opakowanie granicą błędów najwyższego poziomu komponenty odpowiedzialne za routing aplikacji, aby wyświetlić informację: "Coś poszło nie tak", tak jak ma to często miejsce w frameworkach po stronie serwera. Można również opakować pojedyncze fragmenty aplikacji, aby uchronić jej pozostałą część od błędów.

Reviewer: ```suggestion
To jak bardzo szczegółowo zostanie pokryty kod za pomocą granic błędów jest kwestią preferencji. Możliwe jest na przykład opakowanie granicą błędów komponentu najwyższego poziomu odpowiedzialnego za routing aplikacji, aby wyświetlić informację: "Coś poszło nie tak", tak jak ma to często miejsce w frameworkach po stronie serwera. Można również opakować pojedyncze fragmenty, aby uchronić pozostałą część aplikacji od wystąpienia awarii.
```

Reviewer 2: ```suggestion
To, jak bardzo szczegółowo zostanie pokryty kod za pomocą granic błędów, jest kwestią preferencji. Możliwe jest, na przykład, opakowanie granicą błędów komponentu najwyższego poziomu odpowiedzialnego za routing aplikacji, aby wyświetlić informację: "Coś poszło nie tak" - tak jak ma to często miejsce w frameworkach po stronie serwera. Można również opakować pojedyncze fragmenty aplikacji, aby uchronić jej pozostałe części przed błędami.
```

## Nowe zachowanie nieobsłużonych błędów {#new-behavior-for-uncaught-errors}

> Wprowadzenie granic błędów ma ważne następstwo. **Od Reacta w wersji 16, błędy, które nie zostały obsłużone za pomocą granicy błędów, sprawią, że zostanie odmontowane całe drzewo komponentów Reactowych.**

Reviewer 2: ```suggestion
Wprowadzenie granic błędów ma ważne następstwo. **Od Reacta w wersji 16, błędy, które nie zostały obsłużone za pomocą granicy błędów, spowodują odmontowanie całego drzewa komponentów.**
```

> Omawialiśmy tą zmianę i z naszego doświadczenia wynika, że lepiej jest usunąć całe drzewo komponentów niż wyświetlać zepsute fragmenty UI. Na przykład w produkcie takim jak Messenger pozostawienie wyświetlonego zepsutego kawałka UI może sprawić, że ktoś nieświadomie wyśle wiadomość do innej osoby. Również w aplikacjach finansowych wyświetlanie złego stanu konta jest gorszą sytuacją niż niewyświetlenie niczego.

Reviewer: ```suggestion
Omawialiśmy tą zmianę i z naszego doświadczenia wynika, że lepiej jest usunąć całe drzewo komponentów niż wyświetlać zepsute fragmenty UI. Na przykład w produkcie takim jak Messenger pozostawienie wyświetlonego zepsutego kawałka UI może sprawić, że ktoś nieświadomie wyśle wiadomość do innej osoby. Również w aplikacjach finansowych wyświetlanie złego stanu konta jest gorszą sytuacją niż wyświetlenie niczego.
```

> Omawialiśmy tą zmianę i z naszego doświadczenia wynika, że lepiej jest usunąć całe drzewo komponentów niż wyświetlać zepsute fragmenty UI. Na przykład w produkcie takim jak Messenger pozostawienie wyświetlonego zepsutego kawałka UI może sprawić, że ktoś nieświadomie wyśle wiadomość do innej osoby. Również w aplikacjach finansowych wyświetlanie złego stanu konta jest gorszą sytuacją niż niewyświetlenie niczego.

Reviewer 2: ```suggestion
Przedyskutowaliśmy tę zmianę i z naszego doświadczenia wynika, że lepiej jest usunąć całe drzewo komponentów niż wyświetlać zepsute fragmenty UI. Na przykład, w produkcie takim jak Messenger pozostawienie wyświetlonego zepsutego kawałka UI może sprawić, że ktoś nieświadomie wyśle wiadomość do innej osoby. Również w aplikacjach finansowych wyświetlanie złego stanu konta jest gorszą sytuacją niż nie wyświetlenie niczego.
```

> Ta zmiana oznacza, że wraz z migracją do nowej wersji Reacta odkryte zostaną potencjalne błędy w aplikacjach, które do tej pory nie zostały zauważone. Dodanie granic błędów zapewni lepszy user experience gdy coś pójdzie nie tak.

Reviewer: ```suggestion
Ta zmiana oznacza, że wraz z migracją do nowej wersji Reacta odkryte zostaną potencjalne błędy w aplikacjach, które do tej pory nie zostały zauważone. Dodanie granic błędów zapewni lepsze doświadczenie dla użytkownika gdy coś pójdzie nie tak.
```

> Ta zmiana oznacza, że wraz z migracją do nowej wersji Reacta odkryte zostaną potencjalne błędy w aplikacjach, które do tej pory nie zostały zauważone. Dodanie granic błędów zapewni lepszy user experience gdy coś pójdzie nie tak.

Reviewer 2: ```suggestion
Ta zmiana oznacza, że wraz z migracją do nowej wersji Reacta odkryte zostaną błędy w aplikacjach, które do tej pory nie zostały zauważone. Dodanie granic błędów zapewni lepsze doświadczenie dla użytkownika, gdy coś pójdzie nie tak.
```

Przykładowo, Facebook Messenger opakowuje w osobne granice błędów następujące fragmenty aplikacji: zawartość paska bocznego, panel z informacjami o konwersacji, listę konwersacji i pole tekstowe na wiadomość. Jeżeli jeden z tych elementów zadziała nieprawidłowo, reszta pozostanie interaktywa i działająca.

Zachęcamy również do używania (lub zbudowania własnego) narzędzia do raportowania błędów, dzięki czemu będzie możliwe poznanie nieobsłużonych błędów występujących w środowisku produkcyjnym.

## Ślad stosu komponentów {#component-stack-traces}

> React 16, w środowisku deweloperskim, wyświetla w konsoli wszystkie błędy złapane podczas renderowania, nawet jeżeli aplikacja przypadkowo je przejmie. Oprócz wiadomości błędu i javascriptowego stosu dostępny jest również stos komponentów. Dzięki temu wiadomo, gdzie dokładnie w drzewie komponentów wystąpił błąd:

Reviewer 2: ```suggestion
React 16 w środowisku deweloperskim wyświetla w konsoli wszystkie błędy złapane podczas renderowania, nawet jeżeli aplikacja przypadkowo je przejmie. Oprócz wiadomości błędu i javascriptowego stosu, dostępny jest również stos komponentów. Dzięki temu wiadomo, gdzie dokładnie w drzewie komponentów wystąpił błąd:
```

> React 16, w trakcie developmentu, wyświetla w konsoli wszystkie błędy złapane podczas renderowania, nawet jeżeli aplikacja przypadkowo je przejmie. Oprócz wiadomości błędu i stosu JavaScriptu dostępny jest również stos komponentów. Dzięki temu widoczne jest gdzie dokładnie w drzewie komponentów wystąpił błąd:

Reviewer 2: ```suggestion
React 16, w środowisku deweloperskim, wyświetla w konsoli wszystkie błędy złapane podczas renderowania, nawet jeżeli aplikacja przypadkowo je przejmie. Oprócz wiadomości błędu i javascriptowego stosu dostępny jest również stos komponentów. Dzięki temu wiadomo, gdzie dokładnie w drzewie komponentów wystąpił błąd:
```

> <img src="../images/docs/error-boundaries-stack-trace.png" style="max-width:100%" alt="Błąd złapany w komponencie będącym Granicą błędów">

Reviewer 2: ```suggestion
<img src="../images/docs/error-boundaries-stack-trace.png" style="max-width:100%" alt="Błąd złapany w komponencie będącym granicą błędów">
```

> W drzewie komponentów widoczne są również numery linii i nazwy plików. Ten mechanizm domyślnie działa w aplikacjach stworzyonych przy użyciu [Create React App](https://github.com/facebookincubator/create-react-app):

Reviewer 2: ```suggestion
W drzewie komponentów widoczne są również numery linii i nazwy plików. Ten mechanizm domyślnie działa w aplikacjach stworzonych przy użyciu [Create React App](https://github.com/facebookincubator/create-react-app):
```

> <img src="../images/docs/error-boundaries-stack-trace-line-numbers.png" style="max-width:100%" alt="Błąd złapany w komponencie będącym Granicą błędów wraz z numerami linii">

Reviewer 2: ```suggestion
<img src="../images/docs/error-boundaries-stack-trace-line-numbers.png" style="max-width:100%" alt="Błąd złapany w komponencie będącym granicą błędów wraz z numerami linii">
```

> Jeżeli nie używasz Create React App, możesz ręcznie dodać [ten plugin](https://www.npmjs.com/package/babel-plugin-transform-react-jsx-source) do swojej konfiguracji Babela. Jest on stworzony do użycia tylko podczas developmentu i **powinien zostać wyłączony w środowisku produkcyjnym**

Reviewer 2: ```suggestion
Jeżeli nie używasz Create React App, możesz ręcznie dodać [ten plugin](https://www.npmjs.com/package/babel-plugin-transform-react-jsx-source) do swojej konfiguracji Babela. Został on stworzony do używania tylko podczas dewelopmentu i **powinien zostać wyłączony w środowisku produkcyjnym**
```

> Jeżeli nie używasz Create React App, możesz ręcznie dodać [ten plugin](https://www.npmjs.com/package/babel-plugin-transform-react-jsx-source) do swojej konfiguracji Babela. Został on stworzony do używania tylko podczas dewelopmentu i **powinien zostać wyłączony w środowisku produkcyjnym**

Reviewer 2: ```suggestion
Jeżeli nie używasz Create React App, możesz ręcznie dodać [ten plugin](https://www.npmjs.com/package/babel-plugin-transform-react-jsx-source) do swojej konfiguracji Babela. Został on stworzony do używania tylko podczas fazy deweloperskiej i **powinien zostać wyłączony w środowisku produkcyjnym**
```

> Uwaga
>
> Nazwy komponentów wyświetlane w śladzie stosu zależą od własności [`Function.name`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name). Jeżeli obsługujesz starsze przeglądarki, które nie dostarczają jej natywnie (np. IE 11), możesz dodać łatkę taką jak [`function.name-polyfill`](https://github.com/JamesMGreene/Function.name). Alternatywą jest zadeklarowanie wprost [`displayName`](/docs/react-component.html#displayname) we wszystkich komponentach.

## A co z try/catch? {#how-about-trycatch}

> `try` / `catch` jest świetnym rozwiązaniem ale działa tylko dla imperatywnego kodu:

Reviewer 2: ```suggestion
`try` / `catch` jest świetnym rozwiązaniem, ale działa tylko dla imperatywnego kodu:
```

}
```

> Jednakże komponenty Reacta są deklaratywne i określają *co* powinno być wyrenderowane:

Reviewer 2: ```suggestion
Natomiast komponenty reactowe są deklaratywne i określają, *co* powinno zostać wyrenderowane:
```

```js
<Button />
```

> Granice błędów wspierają deklaratywną naturę React'a. Na przykład, jeżeli w metodzie `componentDidUpdate` wystąpi błąd podczas aktualizacji stanu, aplikacja poprawnie zpropaguje błąd do najbliższej granicy błędów.

Reviewer: ```suggestion
Granice błędów zachowują deklaratywną naturę React'a. Na przykład, jeżeli w metodzie `componentDidUpdate` wystąpi błąd podczas aktualizacji stanu, aplikacja poprawnie przekaże błąd do najbliższej granicy błędów.
```

Reviewer 2: ```suggestion
Granice błędów zachowują deklaratywną naturę Reacta. Na przykład, jeżeli w metodzie `componentDidUpdate` wystąpi błąd podczas aktualizacji stanu, aplikacja poprawnie przekaże błąd do najbliższej granicy błędów.
```

> ## A co z uchwytami zdarzeń? {#how-about-event-handlers}

Reviewer 2: ```suggestion
## A co z procedurami obsługi zdarzeń? {#how-about-event-handlers}
```

> Granice błędów nie obsługują błędów wewnątrz uchwytów zdarzeń.

Reviewer 2: ```suggestion
Granice błędów nie obsługują błędów powstałych wewnątrz procedur obsługi zdarzeń.
```

> React nie potrzebuje granic błędów żeby obsłużyć błędy pochodzące z uchwytów zdarzeń. W przeciwieństwie do metod cyklu życia komponentu lub renderu, uchwyty zdarzeń nie są wywoływane w trakcie renderowania. To powoduje, że nawet w przypadku błędu React wie co wyświetlić na ekranie.

Reviewer 2: ```suggestion
React nie potrzebuje granic błędów do przywrócenia aplikacji po błędzie powstałych w procedurze obsługi zdarzeń.  W przeciwieństwie do metod cyklu życia komponentu lub metody renderującej, procedury obsługi zdarzeń nie są wywoływane w trakcie renderowania. Dzięki temu nawet w przypadku błędu React wie, co wyświetlić na ekranie.
```

> Aby obsłużyć błąd w uchwycie zdarzeń należy użyć JavaScriptowego `try` / `catch`:

Reviewer 2: ```suggestion
Aby obsłużyć błąd w procedurze obsługi zdarzenia, należy użyć javascriptowego `try` / `catch`:
```

}
```

Powyższy przykład prezentuje normalne zachowanie JavaScriptu i nie używa granic błędów.

## Zmiany nazewnictwa od Reacta w wersji 15 {#naming-changes-from-react-15}

> React 15 zawierał bardzo okrojoną obsługę granic błędów za pomocą metody o nazwie: `unstable_handleError`. Ta metoda nie jest już obsługiwana i należy zmienić jej nazwę na `componentDidCatch` począwszy od pierwszych beta wersji Reacta 16.

Reviewer 2: ```suggestion
React 15 zawierał bardzo okrojoną obsługę granic błędów za pomocą metody o nazwie `unstable_handleError`. Ta metoda nie jest już obsługiwana i należy zmienić jej nazwę na `componentDidCatch` począwszy od pierwszych beta wersji Reacta 16.
```

> Ze względu na tą zmianę stworzyliśmy [codemod](https://github.com/reactjs/react-codemod#error-boundaries), aby automatycznie dostosować kod.

Reviewer 2: ```suggestion
Ze względu na tę zmianę stworzyliśmy [codemod](https://github.com/reactjs/react-codemod#error-boundaries), który automatycznie przekształci twój kod.
```
